Dockerfile

```diff
@@ -14,35 +14,24 @@ RUN apk --no-cache add \
     build-base \
     git \
     nodejs \
-    npm \
-    && npm install -g pnpm@10 \
-    && rm -rf /var/cache/apk/*
+    pnpm
 
-# Setup repo
-COPY . ${GOPATH}/src/code.gitea.io/gitea
 WORKDIR ${GOPATH}/src/code.gitea.io/gitea
+COPY --exclude=.git/ . .
 
 # Checkout version if set
-RUN if [ -n "${GITEA_VERSION}" ]; then git checkout "${GITEA_VERSION}"; fi \
- && make clean-all build
+RUN --mount=type=cache,target=/go/pkg/mod \
+    --mount=type=cache,target="/root/.cache/go-build" \
+    --mount=type=cache,target=/root/.local/share/pnpm/store \
+    --mount=type=bind,source=".git/",target=".git/" \
```

Comment:
As long as `mount bind` works, then the project root can be fully mounted into the builder image, then no `COPY --exclude=.git/ . .` is needed?

Reply:
Hmm yes, `node_modules` seems to be a problem. Maybe we can leave a comment here to explain why sometimes "copy" sometimes "mount .git"

Reply:
No, keep pnpm installed from apk, that's fine and preferred over installing a useless npm with it.